src/Idea.hs

```diff
@@ -97,9 +97,17 @@ idea' severity hint from to =
 suggest = idea Suggestion
 warn = idea Warning
 ignore = idea Ignore
+ignore' :: (GHC.HasSrcSpan a, Outputable.Outputable a) =>
+           String -> a -> a -> [Refactoring R.SrcSpan] -> Idea
 ignore' = idea' Ignore
 
 ideaN severity hint from to = idea severity hint from to []
+ideaN' :: (GHC.HasSrcSpan a, Outputable.Outputable a) =>
+          Severity -> String -> a -> a -> Idea
+ideaN' severity hint from to = idea' severity hint from to []
 
 suggestN = ideaN Suggestion
+suggestN' :: (GHC.HasSrcSpan a, Outputable.Outputable a) =>
```

Comment:
Not sure where the `'` should go on these.

Reply:
I also added type signatures because it's very annoying to track the argument types every time. Is this fine?

Reply:
Agreed, I think the lack of type signatures before wasn't a great idea (it's just copying the style for `ideaN` etc, where the lack of type signatures was also a bad idea). Adding them is definitely positive.

Reply:
The existence of `'` versions is just "a bit like HSE, but for GHC" - i wouldn't take its placement or semantics too seriously. It probably wants cleaning up after we have finished the migration.